### PR TITLE
Fix some test flakes - use unique name for workloads in STs

### PR DESCRIPTION
**Pull request by caseydavenport**
## Description

This is a hopeful fix for a number of test flakes seen recently with the symptom "pod does not exist" appearing unexpectedly.

The cause is that re-use of the pod name causes race conditions where the same pod, created by a previous test, can still exist in the API at the time that the next test starts (i.e., when the test checks if it needs to create the pod), but be deleted by the time that the test actually goes to query the pod.

By using different names for each test, we remove the possibility that the pod resources will be shared across tests and thus resolve the issue with test ordering. 

An alternative fix might be to wait for pods to be deleted before continuing on to the next test, but that adds unnecessary wait time and inter-test dependency where there need not be any. 

## Todos

- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
None required
```

### CODE REVIEW

File: cni-plugin/tests/calico_cni_k8s_test.go [hunk 1]
```diff
@@ -88,7 +88,8 @@ func ensurePodDeleted(clientset *kubernetes.Clientset, ns string, podName string
 		podName,
 		metav1.DeleteOptions{
 			PropagationPolicy:  &fg,
-			GracePeriodSeconds: &zero})
+			GracePeriodSeconds: &zero,
```
Comment (caseydavenport) [inline cni-plugin/tests/calico_cni_k8s_test.go:91]: whoops looks like my formatter "fixed" a bunch of improperly formatted lines :sweat_smile:

**Comment by neiljerram** [comment]
@caseydavenport Could you add a rationale, for posterity, in case someone in future sees a need for reusing workload names?

**Comment by caseydavenport** [comment]
@neiljerram yep! Updated the description.

File: cni-plugin/tests/calico_cni_k8s_test.go [hunk 2]
```diff
@@ -2542,8 +2545,10 @@ var _ = Describe("Kubernetes CNI tests", func() {
 
 			clientset := getKubernetesClient()
 
-			// Now create a K8s pod.
-			name := "mypod-1"
+			// Create two k8s pods - for this test we want to ensure that the names for the pods
+			// look alike so make sure we handle pods with very similar names.
```
Comment (neiljerram) [inline cni-plugin/tests/calico_cni_k8s_test.go:2549]: nit: so -> to ?

File: cni-plugin/tests/calico_cni_k8s_test.go [hunk 3]
```diff
@@ -3099,7 +3105,7 @@ var _ = Describe("Kubernetes CNI tests", func() {
 			clientset := getKubernetesClient()
 
 			// Now create a K8s pod.
-			name := "mypod-1"
+			name := fmt.Sprintf("pod-%d", rand.Uint32())
```
Comment (neiljerram) [inline cni-plugin/tests/calico_cni_k8s_test.go:3108]: 1. Make a utility function to encapsulate the pod name generation?
2. I suppose it's astronomically unlikely, but in theory we can still get duplicate names with this approach.  Did you consider something like an incrementing index?

**Review by neiljerram — APPROVED** [review]
Some nits and queries, but LGTM.

**Pull request merged**